lib/scripts/configAuditor.js

```diff
@@ -47,78 +44,555 @@ function configAudit (appDir) {
+              delete jsParam[jsKey]
+              break
+            case 'compiler':
+              checkMissing(jsParam, jsKey, keyStack)
+              delete jsParam[jsKey]
+              break
+            case 'whitelist':
+              checkMissing(jsParam, jsKey, keyStack)
+              delete jsParam[jsKey]
+              break
+            case 'blacklist':
+              delete jsParam[jsKey]
+              break
+            case 'output':
+              delete jsParam[jsKey]
+              break
+            case 'symlinkToPublic':
+              delete jsParam[jsKey]
+              break
+            case 'bundler':
+              const bundlerParam = jsParam[jsKey] || {}
+              const defaultBundlerParam = defaultJsParam[jsKey]
+              const bundlerKeys = Object.keys(defaultBundlerParam)
+              bundlerKeys.forEach((bundlerKey) => {
+                keyStack.push(bundlerKey)
+                switch (bundlerKey) {
+                  case 'bundles':
+                    delete bundlerParam[bundlerKey]
+                    break
+                  case 'output':
+                    checkMissing(bundlerParam, bundlerKey, keyStack)
+                    delete bundlerParam[bundlerKey]
+                    break
+                  case 'expose':
+                    delete bundlerParam[bundlerKey]
+                    break
+                }
+                keyStack.pop(bundlerKey)
+              })
+              break
+          }
+          keyStack.pop(jsKey)
+        })
+        break
+      case 'publicFolder':
+        checkMissing(userConfig, key, keyStack)
+        delete userConfig[key]
+        break
+      case 'favicon':
+        checkMissing(userConfig, key, keyStack)
+        delete userConfig[key]
+        break
+      case 'staticsSymlinksToPublic':
+        checkMissing(userConfig, key, keyStack)
+        delete userConfig[key]
+        break
+      case 'versionedPublic':
+        checkMissing(userConfig, key, keyStack)
+        delete userConfig[key]
+        break
+      case 'alwaysHostPublic':
+        checkMissing(userConfig, key, keyStack)
+        delete userConfig[key]
+        break
+      case 'routers':
+        checkMissing(userConfig, key, keyStack)
+        delete userConfig[key]
+        break
+    }
+    keyStack.pop(key)
+  })
+
+  /**
+   * Recursive function that checks for extra params and lets the user know where to find them
+   *
+   * @param userConfigParam - extra params in the userConfig
+   * @param defaultConfigParam - object in the defaultConfig at the key specified in the userConfigParam
+   * @param jsonPath - string containing the path to the extra param
+   */
+  function checkExtraParams (userConfigParam, defaultConfigParam, jsonPath) {
+    const params = Object.entries(userConfigParam)
+    params.map(([key, val]) => {
+      if (val instanceof Object && !(Array.isArray(val)) && val !== null && defaultConfigParam[key] !== undefined) {
```

Comment:
May be possibly good to leave a comment for this conditional that this is when we have a further layer of params we want to examine.

Reply:
I will add a comment